tests/httpsign_test.py

```diff
@@ -149,15 +151,30 @@ def test_authenticate_request(gn, gt, tm):
     # check
     for i, test in enumerate(auth_tests):
         print 'Testing Input {}: {}'.format(i, test['input'])
-        testOutput = httpsign.authenticate_request(test['input']['timestamp'],
-                                               test['input']['nonce'],
-                                               test['input']['body'],
-                                               test['input']['signature'],
-                                               http_verb=test['input']['http_verb'],
-                                               http_resource_uri=test['input']['http_resource_uri'],
-                                               headers=test['input']['headers'],
-                                               key=test['input']['key'])
-        assert_equal(test['output'], testOutput)
+
+        # if the request was invalid, expect an AuthenticationException, otherwise
+        # we should have no problems.
+        if test['output'] is False:
+            assert_raises(httpsign.AuthenticationException, httpsign.authenticate_request,
+                test['input']['timestamp'],
+                test['input']['nonce'],
+                test['input']['body'],
+                test['input']['signature'],
+                "2",
+                test['input']['http_verb'],
+                test['input']['http_resource_uri'],
+                test['input']['headers'],
+                test['input']['key'],
+                None)
+        else:
+            httpsign.authenticate_request(test['input']['timestamp'],
+                test['input']['nonce'],
+                test['input']['body'],
+                test['input']['signature'],
+                http_verb=test['input']['http_verb'],
+                http_resource_uri=test['input']['http_resource_uri'],
+                headers=test['input']['headers'],
+                key=test['input']['key'])
```

Comment:
The if branch passes all parameters by position. The else branch passes the last four by key. Can you make them consistent?